restcomm/restcomm.monitoring.service/src/main/java/org/restcomm/connect/monitoringservice/MonitoringService.java

```diff
@@ -156,8 +158,31 @@ public void onReceive(Object message) throws Exception {
     private void onGetCall(Object message, ActorRef self, ActorRef sender) throws ServletParseException {
         GetCall getCall = (GetCall)message;
         String location = getCall.getIdentifier();
+        if (logger.isDebugEnabled()) {
+            logger.debug("MonitoringService onGetCall, location: "+location);
+        }
         if (location != null) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("onGetCall Before: callLocationMap: "+ callLocationMap.toString());
+            }
             ActorRef call = callLocationMap.get(location);
+            if (logger.isDebugEnabled()) {
+                logger.debug("onGetCall After: callLocationMap: "+ callLocationMap.toString());
+            }
+            if(call == null && location.indexOf("@") != -1 && location.indexOf(":") != -1) {
```

Comment:
@gvagenas As described in the comment in the code it's required in case the Contact Header of the INVITE doesn't contain any user part as it is the case for Restcomm SDKs. So yes Contact header of INVITE is different than the Contact header of REGISTER for Restcomm SDKs. We discussed with antonis but changing the SDK will require a lot of additional testing.  This was tested live and working and doesn't have any performance drawback for existing clients